### ISSUE TITLE
Feature proposal: Generate jennifer code from existing go code

### ISSUE THREAD

**Issue by aloder** [dave/jennifer#54]
I wanted to take an existing file like the one in #50 and convert it so that jennifer generates it. Right now I am doing this by hand. I would like a function that takes in a go file and outputs the code that generates that go file using jennifer.

This would discourage workarounds like Op and allow quick generation of static files. 

Similar to #44 
Solves #50

**Comment by dave** [comment]
This would be great, but more appropriate as a standalone utility - e.g. not integrated into the jennifer codebase. Similar to my [html2vecty](https://github.com/dave/html2vecty) utility for doing the same thing with vecty syntax. I definitely won't have time to do this in the next 6 months so feel free to have a go. I'll happily include a link in the readme when it's finished.

**Comment by aloder** [comment]
I agree, I'll look into making it but I have little experience with ASTs. 

The more I think about it the more appealing a tool like this becomes. You can use this tool as a starting point for generating code, filling in the logic and the variables after. Eliminating most of the monotony of writing generated code.

**Comment by dave** [comment]
Yes it would be useful. Feel free to fork html2vecty and use that as a base...

**Comment by aloder** [comment]
Alright just got the project up to a usable standard: https://github.com/aloder/jenjen 

Let me know what you think.

**Comment by dave** [comment]
Wow this looks great! Won't have time to look at it today or tomorrow but Friday I can have a good play.

**Comment by dave** [comment]
p.s. Are you set on the jenjen name? The jennifer project has a [genjen package](https://github.com/dave/jennifer/tree/master/genjen) that generates the API... might be rather confusing...

**Comment by aloder** [comment]
Yea, I noticed that right after I pushed the name. I can change it. I'll start thinking of others.

**Comment by dave** [comment]
Awesome. I took a very quick look at the code and it looks great!

**Comment by aloder** [comment]
Thank you, I will be improving the code and features as I use it in my own products.

Let me know if you have any suggestions.
